src/types.js

```diff
@@ -0,0 +1,9 @@
+export type DOMContainer = Element | Document;
+export type Instance = Element;
+export type TextInstance = Text;
+export type Props = {
+  autoFocus?: boolean,
+  children?: mixed,
+  hidden?: boolean
+};
+export type HostContext = string;
```

Comment:
can we put this flow-typed/ ?

Reply:
`flow-typed` is for third party libs

Reply:
you can put whatever in it, putting globals in it seems to be a fairly common thing (if these are global)

Reply:
Not comfortable with the idea. Is there a problem like this?

Reply:
React already does this.

Reply:
And I'm sorry 😄 You'll hear me saying that a lot. TBH, I unless we have a good reason I wont deviate.

Reply:
I don't really like the idea of treating compile-time type defs files as if they were code and mixing them in with runnable files,. Most repo's i've worked on with flow tend to put globally available types in flow-typed/ or `decls` or in the file that own said type yeah?

Reply:
haha it's ok.

> TBH, unless we have a good reason we shouldn't deviate.

I understand that. However I don't think that makes sense all the time. React's repo does a lot of things optimally for FB or sub-optimally, because it's an enormous old repo. We are starting scratch and have a chance to think through some of these choices. For instance we aren't retrofitting flow into a mature project like React did

Reply:
> For instance we aren't retrofitting flow into a mature project like React did

While I agree with you on this, docs themselves specify that the folder is meant for third party libs that have bad/incomplete types.

Prepack isn't one of the old projects. They maintain a [separate file](https://github.com/facebook/prepack/blob/master/src/types.js) too

Reply:
I really don't like having a javascript file with no javascript in it. I also don't think think a grab bag of types in a file is very helpful either. We should put then in the files that "define" them or they should be made globally available. For the latter the flow docs themselves suggest putting it in flow-typed/ https://flow.org/en/docs/libdefs/creation/#toc-declaring-a-global-type

Reply:
Alright. Not convinced but it doesn't hurt. Adding `flow-typed` and making those types globally available.

Reply:
thanks :)